src/hooks/useFormattedDate.ts

```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect } from 'react';
+import { format } from 'date-fns';
+
+const useFormattedDate = (date: string, formatString: string) => {
+  const [formattedDate, setFormattedDate] = useState('');
+
+  useEffect(() => {
+    if (date) {
+      const dateObject = new Date(date);
+      setFormattedDate(format(dateObject, formatString));
+    }
+  }, [date, formatString]);
```

Comment:
If you dont have anything async in the useEffect, you don't need it. 

https://react.dev/learn/you-might-not-need-an-effect

This should be replaced with useMemo. We can get rid of the whole custom hook. You can abstract functions that would only have a date as input and would have hardcoded formatString. They would be called specifically for their purpose -> formatDate, formatTime, formatDateTime and so on.

Currently, you repeat none of the formatStrings so it is probably unnecessary. When you repeatedly use one of them, we can abstract it.